test/parallel/test-zlib-random-byte-pipes.js

```diff
@@ -141,18 +141,27 @@ class HashStream extends Stream {
   }
 }
 
-for (const [ createCompress, createDecompress ] of [
-  [ zlib.createGzip, zlib.createGunzip ],
-  [ zlib.createBrotliCompress, zlib.createBrotliDecompress ],
-]) {
-  const inp = new RandomReadStream({ total: 1024, block: 256, jitter: 16 });
-  const out = new HashStream();
-  const gzip = createCompress();
-  const gunz = createDecompress();
-
-  inp.pipe(gzip).pipe(gunz).pipe(out);
-
-  out.on('data', common.mustCall((c) => {
-    assert.strictEqual(c, inp._hash, `Hash '${c}' equals '${inp._hash}'.`);
-  }));
-}
+test('random byte pipes', async (t) => {
+  for (const [ createCompress, createDecompress ] of [
+    [ zlib.createGzip, zlib.createGunzip ],
+    [ zlib.createBrotliCompress, zlib.createBrotliDecompress ],
+  ]) {
+    const { promise, resolve } = Promise.withResolvers();
+    const inp = new RandomReadStream({ total: 1024, block: 256, jitter: 16 });
+    const out = new HashStream();
+    const gzip = createCompress();
+    const gunz = createDecompress();
+
+    inp.pipe(gzip).pipe(gunz).pipe(out);
+
+    const onDataCallback = t.mock.fn();
+    onDataCallback.mock.mockImplementation((c) => {
+      assert.strictEqual(c, inp._hash, `Hash '${c}' equals '${inp._hash}'.`);
+    });
+    out.on('data', onDataCallback);
+    out.on('end', resolve);
+
+    await promise;
```

Comment:
can't we `await once(out, 'end')` ?

Reply:
Yes, definitely! Nice catch. Thanks @targos

Reply:
This means trading a Node.js specific API (`common.mustCall()`) with another Node.js specific API (`events.once()`). What's the point?

Reply:
Ok but I don't see how it is an improvement. It doesn't change anything for the sake of the test.